enclave/core/sgx/enter.S

```diff
@@ -105,15 +105,42 @@ oe_enter:
 
     // Abort if SSA[0].GPRSGX.URSP is within the enclave memory range
     cmp %r12, %r9
-    jb .exception_check_pass
+    jb .exception_handler_stack_check
     cmp %r13, %r9
-    jae .exception_check_pass
+    jae .exception_handler_stack_check
     jmp .abort
 
     // Reaching this point implies SSA[0].GPRSGX.RSP is within the enclave
     // memory range so we do not need additional checks.
 
-.exception_check_pass:
+.exception_handler_stack_check:
+    // Stop speculative execution at target of conditional jump
+    lfence
+
+    // Get the exception_handler_stack_check range
+    mov td_exception_handler_stack(%r11), %r14
+    mov td_exception_handler_stack_size(%r11), %r15
+    add %r14, %r15
+    jc .exception_stack_setup // check for overflow
+
+    // Check if the stack range is within the enclave memory range
+    // If the check fails, fallback to the default behavior (i.e.,
+    // re-using the stack pointer saved in the SSA)
+    cmp %r12, %r14
+    jb .exception_stack_setup
+    cmp %r13, %r15
```

Comment:
Document/describe the scenario where size has been set but stack has not yet been set. That is, when r14 is zero and r15 is 0 + size.

Reply:
now we check size != 0, no longer needs this